src/trio/_tests/test_timeouts.py

```diff
@@ -169,7 +180,79 @@ async def test_timeouts_raise_value_error() -> None:
     ):
         with pytest.raises(
             ValueError,
-            match="^(duration|deadline|timeout) must (not )*be (non-negative|NaN)$",
+            match="^(duration|deadline|timeout|relative deadline) must (not )*be (non-negative|NaN)$",
         ):
```

Comment:
I think I'd just update the error messages to refer to the argument name in all cases, using backticks to make it clear that this is an argument name if needed (e.g. maybe for "seconds"?)

Reply:
Since both `move_on_after` and `CancelScope` want to validate the same input parameter but currently have a different name for it, that would mean one of these has to be true:
* `move_on_after(seconds=-1)` gives `"relative_deadline" can't be negative`
* we duplicate validation code between `move_on_after` and `CancelScope.__attrs_post_init__`
* `move_on_after` catches and reraises the exception from `CancelScope` solely to replace the parameter name in the error message.

Reply:
Duplicating validation logic seems least bad of these options 🤔

Reply:
There is another option we could do. Given how many names this parameter has, we could just add an additional parameter to `CancelScope` itself, with the name to use when reporting errors. Maybe that clutters the API too much though. But it does seem anyone else wrapping might also want to change the error reporting.

Reply:
that sounds quite overcomplicated imo, esp since that would probably include both `deadline` and `relative_deadline`. If users are wrapping it and want pretty errors I don't think it's unreasonable to expect them to catch and reraise